scripts/helmer.sh

```diff
@@ -0,0 +1,206 @@
+    echo CHARTS_LOCATION=\"$CHARTS_LOCATION\"
+    echo VERSION=\"$VERSION\"
+    # Extra parameters
+    echo CRDS_ARGS=\"$CRDS_ARGS\"
+    echo CONTROLLER_ARGS=\"$CONTROLLER_ARGS\"
+    echo DEFAULTS_ARGS=\"$DEFAULTS_ARGS\"
+    # Version map
+    for key in "${!vMap[@]}"; do
+        echo vMap[$key]=\"${vMap[$key]}\"
+    done
+}
+
+load_env() { source /tmp/helmer.env; }
+
+# Parse app $VERSION into chart versions (vMap)
+declare -A vMap
+make_version_map() {
+    # Use tempfile to avoid handling quotes in json > bash
+    local tempfile=$(mktemp -p "${BATS_RUN_TMPDIR:-/tmp}" vermap-XXXXX.json)
+
+    if [ "$VERSION" != local ]; then
+        # Do single helm search query to speed up the process
+        helm repo update $REPO_NAME --fail-on-repo-update-fail >/dev/null
+        helm search repo --fail-on-no-result $REPO_NAME/ --versions --devel -o json \
+            | jq -ec --arg appv "$VERSION" '
+                def remap(items): items | unique_by(.name) | map({(.name): .version}) | add;
+                unique_by(.name) as $latest |
+                {
+                    appv: remap(map(select(.app_version == $appv))),
+                    next: remap($latest),
+                    prev: remap(map(select(.app_version != $latest[0].app_version and (.app_version | contains("-rc") | not)))),
+                }' > "$tempfile"
+    fi
+
+    # Load $VERSION from json into vMap array
+    for chart in crds controller defaults; do
+        case $VERSION in
+            # Next: last version from helm search kubewarden --devel
+            next)    vMap["$chart"]=$(jq -er --arg c $chart '.next["kubewarden/kubewarden-" + $c]' "$tempfile") ;;
+            # Prev: previous stable version (older than next and not -rc)
+            prev)    vMap["$chart"]=$(jq -er --arg c $chart '.prev["kubewarden/kubewarden-" + $c]' "$tempfile") ;;
+            # App: Exact helm app version: v1.17.0-rc2
+            v[1-9]*) vMap["$chart"]=$(jq -er --arg c $chart '.appv["kubewarden/kubewarden-" + $c]' "$tempfile") ;;
+            # Local: directory with kw charts (kubewarden-crds, kubewarden-controller, kubewarden-defaults)
+            local)   vMap["$chart"]=$(helm show chart $CHARTS_LOCATION/kubewarden-$chart | yq '.version') ;;
+            *) echo "Bad VERSION: ${VERSION:-}"; exit 1;;
+        esac
+    done
+    rm "$tempfile"
+
+    # Save initial settings
+    print_env > /tmp/helmer.env
+}
+
+# ==================================================================================================
+# Install & Upgrade kubewarden (change chart version)
+
+# Usage: helm_in kubewarden-crds [--params ..]
+helm_in() { helm install --wait --wait-for-jobs --namespace $NAMESPACE "${@:2}" "$1" "$CHARTS_LOCATION/$1"; }
+
+# Install selected $VERSION
+do_install() {
+    echo "Install $VERSION: ${vMap[*]}"
+    # Cert-manager is required by Kubewarden <= v1.16.0 (crds < 1.9.0)
+    if printf '%s\n' "${vMap[crds]}" "1.8.9" | sort -V -C; then
```

Comment:
I wonder if with v1.17.0 out, we can remove the cert-manager install. Doesn't hurt to have it though.